src/octoprint/plugins/virtual_printer/__init__.py

```diff
@@ -193,7 +201,7 @@ def _copy_to_incoming(self):
                             del buffered[: termpos + termlen]
                             received = bytes(line)
 
-                            print(f"<<< {received.rstrip()}")
+                            print(f"<<< {received.rstrip().decode('utf-8')}")
```

Comment:
`to_str`

Reply:
I assume you mean `to_unicode`? `to_str` is marked as deprecated and suggests to use `to_bytes` which encodes.

Reply:
Er, yes... sorry, got confused there.